### PR TITLE
Improvements to Fixtures

### PR DESCRIPTION
**Pull request by athornz**
This PR improves Fixtures so they can be used with a custom Dispatcher.

The creation of a MockResponse from a fixture is now done via a method on the fixture itself.

This allows a fixture to be used like:

```
Fixture.parseFrom(Fixtures.SIMPLE).toMockResponse();
```

The DelayedDispatcher has been removed and the delay method on a MockResponse is used instead. It seemed like duplicated logic, and the delay would not work when using a custom Dispatcher.

A couple of dependencies have been updated, and the build tweaked to work with the latest AGP plugin.

### CODE REVIEW

**Comment by athornz** [comment]
@orhanobut Have you had a chance to review this at all?

**Comment by orhanobut** [comment]
@athornz I'm on it. Need to confirm a couple of things. Thanks for PR btw.

File: mockwebserverplus/src/main/java/com/orhanobut/mockwebserverplus/Fixture.java [hunk 1]
```diff
@@ -74,4 +77,23 @@ private static String readPathIntoString(String path) throws IOException {
 
     return out.toString();
   }
-}
+
+  public MockResponse toMockResponse() {
```
Comment (orhanobut) [inline mockwebserverplus/src/main/java/com/orhanobut/mockwebserverplus/Fixture.java]: 👍

File: sample/build.gradle [hunk 2]
```diff
@@ -63,17 +63,14 @@ task generateFixtures(group: 'fixtures') << {
   new File(path).write(builder.toString())
 }
 
-afterEvaluate {
-  generateDebugSources.dependsOn generateFixtures
-  generateReleaseSources.dependsOn generateFixtures
-}
+preBuild.dependsOn generateFixtures
```
Comment (orhanobut) [inline sample/build.gradle]: 👍

File: mockwebserverplus/src/main/java/com/orhanobut/mockwebserverplus/DelayedDispatcher.java [hunk 3]
```diff
@@ -1,26 +0,0 @@
...
-import java.util.logging.Logger;
-
-import okhttp3.mockwebserver.MockResponse;
-import okhttp3.mockwebserver.QueueDispatcher;
-import okhttp3.mockwebserver.RecordedRequest;
-
-/**
- * Delays {@link okhttp3.mockwebserver.MockWebServer} response for specified seconds.
- */
-public class DelayedDispatcher extends QueueDispatcher {
```
Comment (orhanobut) [inline mockwebserverplus/src/main/java/com/orhanobut/mockwebserverplus/DelayedDispatcher.java]: MockWebServer.delayBody actually works different than this dispatcher. This was used to simulate connection timeout. `delayBody` request happens immediately and it simulates read timeout. 

```
request -> connection (timeout, delayed dispatcher) -> read (timeout, delayBody)
```

However, with anonymous dispatcher, client can just create an anonymous dispatcher and add the delay themselves.   `Thread.sleep(delayMillis)`. Therefore it's OK to remove it. Less feature, less maintenance 👍

**Comment by orhanobut** [comment]
👍

**Pull request merged**